### PR TITLE
Reduce allocations of Location instances

### PR DESCRIPTION
**Pull request by Youssef1313**
An attempt to reduce the allocations I'm seeing in https://github.com/dotnet/roslyn/issues/68934

### CODE REVIEW

File: src/Compilers/CSharp/Portable/Symbols/Source/SourceEventAccessorSymbol.cs [hunk 1]
```diff
@@ -204,6 +204,8 @@ public sealed override ImmutableArray<ImmutableArray<TypeWithAnnotations>> GetTy
         public sealed override ImmutableArray<TypeParameterConstraintKind> GetTypeParameterConstraintKinds()
             => ImmutableArray<TypeParameterConstraintKind>.Empty;
 
+        public sealed override Location TryGetFirstLocation() => _location;
+
```
Comment (AlekseyTs) [inline src/Compilers/CSharp/Portable/Symbols/Source/SourceEventAccessorSymbol.cs:208]: Why is this override necessary? It looks like ```SourceMemberMethodSymbol``` has the same implementation.

File: src/Compilers/CSharp/Portable/Symbols/Source/SourcePropertySymbolBase.cs [hunk 2]
```diff
@@ -654,6 +654,8 @@ internal SyntaxTree SyntaxTree
             }
         }
 
+        public sealed override Location TryGetFirstLocation() => Location;
```
Comment (AlekseyTs) [inline src/Compilers/CSharp/Portable/Symbols/Source/SourcePropertySymbolBase.cs:657]: >public sealed override Location TryGetFirstLocation() => Location; [](http://example.com/codeflow?start=8&length=66)

Please place this override next to ```public override ImmutableArray<Location> Locations``` to make it obvious why implementation is correct.

**Comment by AlekseyTs** [comment]
@dotnet/roslyn-compiler For the second review.

**Comment by Youssef1313** [comment]
@333fred/@RikkiGibson Could you review please? Thanks!

**Comment by AlekseyTs** [comment]
@dotnet/roslyn-compiler For the second review.

**Review by jcouv — APPROVED** [review]
LGTM Thanks (iteration 5)

**Pull request merged**